batch.go

```diff
@@ -0,0 +1,277 @@
+package anthropic
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type ResultType string
+
+const (
+	ResultTypeSucceeded ResultType = "succeeded"
+	ResultTypeErrored   ResultType = "errored"
+	ResultTypeCanceled  ResultType = "canceled"
+	ResultTypeExpired   ResultType = "expired"
+)
+
+type BatchId string
+
+type BatchResponseType string
+
+const (
+	BatchResponseTypeMessageBatch BatchResponseType = "message_batch"
+)
+
+type ProcessingStatus string
+
+const (
+	ProcessingStatusInProgress ProcessingStatus = "in_progress"
+	ProcessingStatusCanceling  ProcessingStatus = "canceling"
+	ProcessingStatusEnded      ProcessingStatus = "ended"
+)
+
+// While in beta, batches may contain up to 10,000 requests and be up to 32 MB in total size.
```

Comment:
Would probably be premature optimisation to verify this before sending a request.. right?

Reply:
I think so, too